keps/sig-autoscaling/20181106-in-place-update-of-pod-resources.md

```diff
@@ -0,0 +1,376 @@
+     ContainerStatus.ResourcesAllocated values.
+   * _Case 2_: ResizingPod.Status is false, and ResizingPod.Reason is
+     FailedNodeCapacity. It continues account for Node resources using
+     max(ResourceRequirements, ResourcesAllocated), and by default Kubelet is
+     expected to backoff and retry the resize at a later time, or optionally
+     when Pods depart the Node (thus creating room).
+1. The initiating actor observes that ResizingPod.Reason and/or
+   ContainerStatus.ResourcesAllocated fields have changed.
+   * _Case 1_: ResizingPod.Status is false, ResizingPod.Reason is Success,
+     and Pod's Container.ResourceRequirements matches
+     ContainerStatus.ResourcesAllocated, signifying a successful completion of
+     in-place Pod resources resizing.
+   * _Case 2_: ResizingPod.Status is false, and ResizingPod.Reason is
+     FailedNodeCapacity. The initiating actor may take alternative action.
+     For example, based on Retry policy, initiating actor (e.g VPA) may choose to:
+     - Evict the Pod to trigger a replacement Pod with updated resources,
+     - Do nothing, and let Kubelet backoff and retry in-place resize.
+
+#### Transitions of the ResizingPod PodCondition
+
+The following diagram shows possible transitions of ResizingPod.Status and
+ResizingPod.Reason fields respectively.
+
+```text
+
+   +-------------------------------------+
+   |                                     |
+   |                                    2|
+   |                                +----v----+
+   |                                |         |
+   |          +---------------------+  false  +-------------------------+
+   |          |                     | Success |                         |
+   |          |                     |         |                         |
+   |          |                     +----+----+                         |
+   |          |                          |                              |
+   |         1|                         3|                             5|
+   |   +------v-----+             +------v-----+             +----------v---------+
+   |   |            |             |            |             |                    |
+   |   |    true    |4            |    true    |7            |        false       |
+   +---+ InProgress <-------------+ PreEmpting <-------------+ FailedNodeCapacity |
+       |            |             |            |             |                    |
+       +------^-----+             +------------+             +----------+---------+
+             6|                                                         |
+              |                                                         |
+              +---------------------------------------------------------+
+
+```
+
+1. Kubelet, on initiating in-place resize.
+1. Kubelet, on successful completion of in-place resize.
+1. Kubelet, on initiating pre-emption of lower priority Pods.
```

Comment:
this leaks information about the state of the node to a pod.  a pod should just know that it's desire is to make the change, it should not know the impact it is having to other pods on the node.

Reply:
This has been removed from the KEP in updated design. Resolving.